### PR TITLE
Skeleton for ideas on Batch and Task APIs.

### PR DESCRIPTION
**Pull request by sollecitom**
Engine 1348

DO NOT MERGE!

### CODE REVIEW

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/AdiGenerationPolicy.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*
+ * (c) Copyright 2015 EVRYTHNG Ltd London / Zurich
+ * www.evrythng.com
+ */
+package com.evrythng.thng.resource.model.store;
+
+public class AdiGenerationPolicy {
```
Comment (dorogush) [inline thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/AdiGenerationPolicy.java]: Serializable?

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/TaskOnBatch.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/*
+ * (c) Copyright 2015 EVRYTHNG Ltd London / Zurich
+ * www.evrythng.com
+ */
+package com.evrythng.thng.resource.model.store;
+
+public class TaskOnBatch extends Task {
```
Comment (dorogush) [inline thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/TaskOnBatch.java]: abstract?

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/FileBasedAdiInputParameters.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+package com.evrythng.thng.resource.model.store;
+
+/**
+ * TODO write javadoc here
+ */
+public class FileBasedAdiInputParameters extends BaseAdiInputParameters {
+
+	private String location;
+	private Format format;
+	private String shortDomain;
```
Comment (dorogush) [inline thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/FileBasedAdiInputParameters.java]: Move to BaseAdiInputParameters?

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Task.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+/**
+ * Model representation for long <em>tasks</em> to distribute between machines.
+ */
+public class Task extends DurableResourceModel {
+
+	private static final long serialVersionUID = -9117128840439882175L;
+	// TODO _MS_
+	private Progress progress;
+	private ResourceLocation resourceLocation;
```
Comment (dorogush) [inline thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Task.java]: Don't understand how it will be used

**Comment by dorogush** [comment]
We may also need some *Visitor classes to avoid instanceof

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/AdiGenerationPolicy.java (hunk 1 above)
Comment (sollecitom) [inline thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/AdiGenerationPolicy.java]: Works without.

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/TaskOnBatch.java (hunk 2 above)
Comment (sollecitom) [inline thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/TaskOnBatch.java]: Right :) fixed

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/FileBasedAdiInputParameters.java (hunk 3 above)
Comment (sollecitom) [inline thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/FileBasedAdiInputParameters.java]: Makes sense.

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Task.java (hunk 4 above)
Comment (sollecitom) [inline thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Task.java]: Just a stub for now. The idea is that from a Task, you will need to fetch a possible big file from somewhere. This models the somewhere.

**Pull request merged**